packages/signers/README.md

```diff
@@ -0,0 +1,368 @@
+-   **Potential confirmation**. Whilst this is not required by this interface, it is also worth noting that most wallets will also wait for the transaction to be confirmed (typically with a `confirmed` commitment) before notifying the app that they are done.
+
+#### `TransactionSigner<TAddress>`
+
+Union interface that uses any of the available transaction signers.
+
+```ts
+type TransactionSigner<TAddress extends string = string> =
+    | TransactionPartialSigner<TAddress>
+    | TransactionModifyingSigner<TAddress>
+    | TransactionSendingSigner<TAddress>;
+```
+
+### Functions
+
+#### Type guards
+
+Each of the transaction interfaces described above comes with two type guards that allow us to check whether or not a given value is a transaction signer of the requested type. One that returns a boolean and one that asserts by throwing an error if the provided value is not of the expected interface.
+
+```ts
+const myAddress = address('1234..5678');
+
+isTransactionPartialSigner({ address: myAddress, signTransaction: async () => {} }); // ✅ true
+isTransactionPartialSigner({ address: myAddress }); // ❌ false
+assertIsTransactionPartialSigner({ address: myAddress, signTransaction: async () => {} }); // ✅ void
+assertIsTransactionPartialSigner({ address: myAddress }); // ❌ Throws an error.
+
+isTransactionModifyingSigner({ address: myAddress, modifyAndSignTransaction: async () => {} }); // ✅ true
+isTransactionModifyingSigner({ address: myAddress }); // ❌ false
+assertIsTransactionModifyingSigner({ address: myAddress, modifyAndSignTransaction: async () => {} }); // ✅ void
+assertIsTransactionModifyingSigner({ address: myAddress }); // ❌ Throws an error.
+
+isTransactionSendingSigner({ address: myAddress, signAndSignTransaction: async () => {} }); // ✅ true
+isTransactionSendingSigner({ address: myAddress }); // ❌ false
+assertIsTransactionSendingSigner({ address: myAddress, signAndSignTransaction: async () => {} }); // ✅ void
+assertIsTransactionSendingSigner({ address: myAddress }); // ❌ Throws an error.
+
+isTransactionSigner({ address: myAddress, signTransaction: async () => {} }); // ✅ true
+isTransactionSigner({ address: myAddress, modifyAndSignTransaction: async () => {} }); // ✅ true
+isTransactionSigner({ address: myAddress, signAndSignTransaction: async () => {} }); // ✅ true
+assertIsTransactionSigner({ address: myAddress, signTransaction: async () => {} }); // ✅ void
+assertIsTransactionSigner({ address: myAddress, modifyAndSignTransaction: async () => {} }); // ✅ void
+assertIsTransactionSigner({ address: myAddress, signAndSignTransaction: async () => {} }); // ✅ void
+```
+
+## Creating and generating KeyPair signers
+
+### Types
+
+#### `KeyPairSigner<TAddress>`
+
+Defines a signer that uses a `CryptoKeyPair` to sign messages and transactions. It implements both the `MessagePartialSigner` and `TransactionPartialSigner` interfaces and keeps track of the `CryptoKeyPair` instance used to sign messages and transactions.
```

Comment:
Do we not have any reason to roll `KeyPairModifyingSigner` and `KeyPairSendingSigner`?

Reply:
Even just as interfaces someone could implement on their own?

Reply:
I think signers should provide the most convenient interfaces they can provide. In the case of `CryptoKeyPairs`, we can provide the more efficient and flexible `*PartialSigner` interfaces, so I'm not sure why we'd want to offer more than that. Do you have any use-cases in mind?